### PR TITLE
Open acc reduction cleanup

### PR DESCRIPTION
**Pull request by johnmauff**
This PR addresses several bugs associated with the placement of global integrals into GPU memory.  They were moved back to the CPU.  It eliminates a large number of '$acc compare' directives and some commented out stop statements used for debugging.  It did add a new routine used for debugging 'maxmin2dhalo' and added it to several locations in the code.

### CODE REVIEW

File: src/misclibs.F [hunk 1]
```diff
@@ -4014,27 +4018,44 @@ subroutine calcmoe(nstat,rstat,ruh,rvh,rmh,rho,ua,va,wa,qv,ql,qi,vr)
...
+      call MPI_REDUCE(qltmp,var,1,MPI_DOUBLE_PRECISION,MPI_SUM,0,  &
+                      MPI_COMM_WORLD,ierr)
+      qltmp=var
+
+      var=0.0d0
+      call MPI_REDUCE(vrtmp,var,1,MPI_DOUBLE_PRECISION,MPI_SUM,0,  &
+                      MPI_COMM_WORLD,ierr)
+      vrtmp=var
+
       if(myid.eq.0)then
```
Comment (sjsprecious) [inline src/misclibs.F:4230]: Are these for diagnostic only and will be removed later?

File: src/kessler.F [hunk 2]
```diff
@@ -803,14 +813,16 @@ subroutine fallout_GPU(dt,train,ruh,rvh,zh,mh,mf,rain,prate,rr,rho,   &
       enddo
     enddo
     enddo
-
     !$acc end parallel
+
+    !JMD FIXME:  This loop looks incorrect. what is going on with the n-loop
```
Comment (sjsprecious) [inline src/kessler.F:818]: How do you know that this loop is incorrect?

File: src/misclibs.F (hunk 1 above)
Comment (johnmauff) [reply src/misclibs.F:4230]: Yes, these are for diagnostic purposes only.  I wanted to identify which parts of the TMW variable was problematic.

File: src/kessler.F (hunk 2 above)
Comment (johnmauff) [reply src/kessler.F:818]: This is a note to myself to come back and look at this code in more detail.  Note that the currently dx=40m namelist does not activate this code.

Comment (sjsprecious) [reply src/kessler.F:818]: Thanks for the explanation.

**Review by sjsprecious — APPROVED** [review]

**Pull request merged**